### PR TITLE
deprecation(semver): `Comparator` overloads for `comparatorMin()` and `comparatorMax()`

### PR DESCRIPTION
**Pull request by timreichen**
- adds `Comparator` overloads to `comparatorMin()` and `comparatorMax()`
- deprecates other overloads

Reason being that it is unintuitive to spread `Comparator` arguments over a comparator function.

### CODE REVIEW

File: semver/comparator_max.ts [hunk 1]
```diff
@@ -1,15 +1,29 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
-import type { Operator, SemVer } from "./types.ts";
+import type { Comparator, Operator, SemVer } from "./types.ts";
 import { ANY, INVALID, MAX } from "./constants.ts";
+import { isComparator } from "./is_comparator.ts";
 
+/**
+ * @deprecated (will be removed in 0.213.0) Use a {@linkcode Comparator} argument instead.
+ */
+export function comparatorMax(semver: SemVer, operator: Operator): SemVer;
```
Comment (iuioiua) [inline semver/comparator_max.ts:9]: We should still retain the documentation for this overload. Ditto for `comparatorMin()`.

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by iuioiua — APPROVED** [review]
LGTM. Let's see what Yoshiya thinks.

**Comment by kt3k** [comment]
This doesn't make much sense to me as we are now discussing #4047. I think we should first settle on #4047

**Comment by iuioiua** [comment]
Closing in favour of #4131. Thank you, Tim.

**Pull request closed without merging**